src/control/server/storage/scm/ipmctl.go

```diff
@@ -32,20 +34,40 @@ const (
 )
 
 type (
-	runCmdFn   func(string) (string, error)
+	pmemCmd struct {
+		BinaryName string
+		Args       []string
+	}
+	runCmdFn   func(logging.Logger, pmemCmd) (string, error)
 	lookPathFn func(string) (string, error)
 )
 
-func run(cmd string) (string, error) {
-	out, err := exec.Command("bash", "-c", cmd).Output()
+func (pc *pmemCmd) String() string {
+	return fmt.Sprintf("cmd %s, args %v", pc.BinaryName, pc.Args)
```

Comment:
Just a thought but it might be a little clearer to present it as the string that would actually be entered by the user. Something like:

```suggestion
	return fmt.Sprintf("command: %s %s", pc.BinaryName, strings.Join(pc.Args, " "))
```

Reply:
I want to be able to discern the exact string arguments passed to exec.Command i.e. ["-o nvxml"] vs ["-o", "nvxml"] as strangely this has been making a difference in execution.

Reply:
Ah, okay. That makes sense.